src/operations.jl

```diff
@@ -1,33 +1,33 @@
-export synsets, antonyms, hypernyms, hyponyms, expanded_hypernyms
+export synsets, antonyms, hypernyms, hyponyms, expanded_hypernyms, expanded_hyponyms
 
 synsets(db::DB, lemma::Lemma) = map(lemma.synset_offsets) do offset
     db.synsets[lemma.pos][offset]
 end
 
-relation(db::DB, synset::Synset, pointer_sym) = map(
+relation(db::DB, synset::Synset, pointer_sym) = Set(map(
     ptr -> db.synsets[ptr.pos][ptr.offset],
     filter(ptr -> ptr.sym == pointer_sym, synset.pointers)
-)
+))
 
-antonyms(db::DB, synset::Synset)  = relation(db, synset, ANTONYM)
-hyponyms(db::DB, synset::Synset)  = relation(db, synset, HYPONYM)
+function expanded_relation(db::DB, synset::Synset, pointer_sym)
+    all_related = Set{Synset}()
+    queue = relation(db, synset, pointer_sym)
 
-function hypernyms(db::DB, synset::Synset)
-    for ptr in synset.pointers
-        if ptr.sym == HYPERNYM
-            return db.synsets[synset.synset_type][ptr.offset]
-        end
+    while !isempty(queue)
+        node = pop!(queue)
+        push!(all_related, node)
+        local_relatives = setdiff(relation(db, node, pointer_sym), all_related)
+        delete!(local_relatives, synset)
```

Comment:
if you inititalized `all_related` to contain `synset` at the start,
then you would only need to delete it once at the end.

More broadly, though perhaps a it might be better to do

```
function expanded_relation(db::DB, synset::Synset, pointer_sym)
    all_related = Set{Synset}(synset)
    queue = relation(db, synset, pointer_sym)

    while !isempty(queue)
        node = pop!(queue)
        node in all_related && continue  # done this one already 
        push!(all_related, node)

        union!(queue, relation(db, node, pointer_sym))
    end

    delete!(all_related, synset)
    return all_related
end
```

since this avoids allocating a new set in `setdiff`